File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def solve(self, root,summ):

        if root == None: return 0
        summ = summ*10 + root.val

        if root.left is None and root.right is None:
            return summ

        left = self.solve(root.left,summ)
        right = self.solve(root.right,summ)

        return left + right
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        summ = 0
        return self.solve(root,summ)

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int solve(TreeNode* root, int sum){

        if(root == nullptr) return 0;

        sum = sum*10 + root->val;

        if (root->left == nullptr && root->right == nullptr){
            return sum;
        }

        int left = solve(root->left, sum);
        int right = solve(root->right, sum);

        return left + right;
    }
    int sumNumbers(TreeNode* root) {
        return solve(root,0);
    }
};